src/Particle/ParticleSet.cpp

```diff
@@ -517,6 +517,7 @@ void ParticleSet::mw_computeNewPosDistTablesAndSK(const RefVectorWithLeader<Part
       const auto dt_list(extractDTRefList(p_list, i));
       p_leader.DistTables[i]->mw_move(dt_list, p_list, new_positions, iat, maybe_accept);
     }
+    PRAGMA_OFFLOAD("omp taskwait")
```

Comment:
I think document what is being waited on, `omp taskwait` is completely generic.  I'm pretty sure it just the transfer to host (for some combinations of DT_options) of distance table values calculated on the device. It's also possible the the distance tables won't be needed for some time and if so there is no need to wait here,  that optimization will be easy to see if this is documented.  Does the SK->makeMove call depend on these tasks?  It doesn't seem to, can the pragma be moved after it?

Reply:
I added in-place documentation about the wait and also added explanation about asynchronous behavior of DistanceTableData::mw_move. The `SK->makeMove` question, probably better to put taskwait after but we don't really care, I'm not ware any production feature activates that. So I will just leave it there.